### DISCUSSION TITLE
Topic search in GraphQL misses forks

### DISCUSSION THREAD

**Discussion by mihai-sysbio** [community/community#24507] [Apps, API and Webhooks]
<p>I’m trying to replicate the results of topic search, specifically <a href="https://github.com/topics/standard-gem" rel="nofollow noopener">https://github.com/topics/standard-gem</a> . The GraphQL query results doesn’t seem to include repository forks, ie <code>mihai-sysbio/Human-GEM</code>:</p>
<pre><code class="lang-auto">{
  search(first: 100, query: "topic:standard-GEM", type:REPOSITORY) {
    repos: edges {
      repo: node {
        ... on Repository {
          nameWithOwner
        }
      }
    }
  }
}
</code></pre>
<p>Is there a way to specify that I’d like the search results to include forks, besides querying for them?</p>

**Comment by francisfuzz** [comment, marked as answer]
<aside class="quote no-group" data-username="mihai-sysbio" data-post="1" data-topic="122719">
<div class="title">
<div class="quote-controls"></div>
 mihai-sysbio:</div>
<blockquote>
<pre><code class="lang-auto">{
  search(first: 100, query: "topic:standard-GEM", type:REPOSITORY) {
    repos: edges {
      repo: node {
        ... on Repository {
          nameWithOwner
        }
      }
    }
  }
}
</code></pre>
</blockquote>
</aside>
<p>:wave: hello there <a class="mention" href="/mihai-sysbio">@mihai-sysbio</a>, and welcome to the GitHub Support Community! :tada:</p>
<p>To include forks in the search results, you will need to add  <code>fork:true</code>  or  <code>fork:only</code>  to your query (source: <a href="https://docs.github.com/en/github/searching-for-information-on-github/searching-for-repositories"><code>Searching for repositories</code></a>).</p>
<p>Here’s an example query that includes forks in the search results:</p>
<pre><code class="lang-auto">{
  search(first: 100, query: "fork:true topic:standard-GEM", type:REPOSITORY) {
    repos: edges {
      repo: node {
        ... on Repository {
          nameWithOwner
        }
      }
    }
  }
}
</code></pre>
<p>At this time of writing, here’s the result set:</p>
<pre><code class="lang-json">{
  "data": {
    "search": {
      "repos": [
        {
          "repo": {
            "nameWithOwner": "SysBioChalmers/Human-GEM"
          }
        },
        {
          "repo": {
            "nameWithOwner": "MetabolicAtlas/standard-GEM"
          }
        },
        {
          "repo": {
            "nameWithOwner": "mihai-sysbio/Human-GEM"
          }
        }
      ]
    }
  }
}
</code></pre>
<p>From <a href="https://github.com/MetabolicAtlas/standard-GEM/network/members" class="inline-onebox">Forks · MetabolicAtlas/standard-GEM · GitHub</a>, <code>mihai-sysbio / standard-GEM</code> is a fork of <code>MetabolicAtlas / standard-GEM</code>. I hope this helps!</p>

**Reply by AfsaneMohamady** [reply]
?

**Comment by mihai-sysbio** [comment]
<p>Thank you <a class="mention" href="/francisfuzz">@francisfuzz</a>.<br>
The <code>fork: true</code> / <code>fork: only</code> is easy to miss. Rather than dividing the documentation into two pages (“searching for repositories” and “searching in forks”), I would find it more intuitive to have both on the same page. That’s because I am interpreting the default value of a normal search to be “fork: false”. Also, a fork is to me a still a repository, kind of at the same level of hierarchy, but perhaps the GitHub philosophy is different.</p>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/mihai-sysbio">@mihai-sysbio</a> - You’re welcome! We’re always working to improve GitHub and the GitHub Support Community, and we consider every suggestion we receive.</p>
<p>Would you mind <a href="https://support.github.com/contact/feedback">submitting this through our official product feedback form</a> so that our product team can track your request? That’s the best place to share requests like these in consideration for future iterations of GitHub features. :v:</p>
